src/eventbridge.ts

```diff
@@ -0,0 +1,68 @@
+import { aws_events } from "aws-cdk-lib";
+import { Rule } from "aws-cdk-lib/aws-events";
+import { Construct } from "constructs";
+import { FunctionDecl } from "./declaration";
+import { synthesizeEventPattern } from "./eventpattern";
+
+export interface EventBusEvent<
+  T = any,
+  DetailType extends string = string,
+  Source extends string = string
+> {
+  source: Source;
+  detailType: DetailType;
+  detail: T;
+}
+
+type EventHandlerFunction<E extends EventBusEvent = EventBusEvent<any>> = (
+  event: E
+) => void;
+
+export type EventPredicateFunction<
```

Comment:
What's the difference between this and `EventHandlerFunction`?

Reply:
Handler function will be for the input transformer (to). They may end up the same and I'll remove one.